### PR TITLE
Add support for switching to restricted mode

### PR DESCRIPTION
**Pull request by DTTerastar**
When I try and restore I have enough users fighting for connections that I often lose the race and end up locked out of my db with single user mode on, but I'm not the one!  This patch makes it possible to use restricted mode instead.

### CODE REVIEW

File: src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs [hunk 1]
```diff
@@ -42,12 +34,33 @@ public RestoreSqlBackupSettings()
...
+        /// </summary>
+        public DbUserMode SwitchToUserMode { get; set; } = DbUserMode.SingleUser;
+
         /// <summary>
         /// Before restoring backup, database will be switched to a single user mode. 
         /// Default operation is to go into single user mode. However in some situation this might not work.
         /// Use this switch to bypass single user mode and restore the DB as it is
         /// </summary>
-        public bool SwitchToSingleUserMode { get; set; }
+        public bool SwitchToRestrictedUserMode { get; set; }
```
Comment (trailmax) [inline src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs]: As far as I can see this is not used anywhere and just been introduced. If we don't need it, can we remove it please?

File: src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs [hunk 2]
```diff
@@ -59,5 +72,13 @@ public RestoreSqlBackupSettings()
         /// </summary>
         public int? DifferentialBackupSetFile { get; set; }
 
+        
+    }
+
+    public enum DbUserMode
```
Comment (trailmax) [inline src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs]: Can you please add xml documentation with a brief explanation of what it does

File: src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs (hunk 1 above)
Comment (DTTerastar) [reply src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs]: Fixed

File: src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs (hunk 2 above)
Comment (DTTerastar) [reply src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs]: Done

**Comment by DTTerastar** [comment]
It's failing because I used a newer language feature.  Upgrading cake still has it using MSBuild 14.

**Comment by trailmax** [comment]
@DTTerastar I'm away on holidays now. Back next week, so this will have to wait for a bit. 
Strange one with MSBuild - change of cake version did help witht he build locally. I guess AppVeyor needs a setting changed to use latest MSBuild - I'll dig into that on my return.

**Comment by trailmax** [comment]
Merged it. I'll sort out the build.
Thank you for your effort. Apologies for taking time - life takes over.

**Comment by trailmax** [comment]
@DTTerastar There is a release out now with your work. Currently is getting validated by Nuget, but should be through is a few minutes.

**Comment by DTTerastar** [comment]
I saw it this morning!  Awesome!  Thanks!

**Pull request merged**